java/gandiva/src/main/cpp/jni_common.cc

```diff
@@ -625,8 +682,61 @@ Status make_record_batch_with_buf_addrs(SchemaPtr schema, int num_rows,
       buffers.push_back(offsets);
     }
 
-    auto array_data = arrow::ArrayData::Make(field->type(), num_rows, std::move(buffers));
+
+
+
+auto type = field->type();
+auto type_id = type->id();
+    if (type_id == arrow::Type::LIST) {
+
+            if (buf_idx >= in_bufs_len) {
+        return Status::Invalid("insufficient number of in_buf_addrs");
+      }
+
+      // add offsets buffer for variable-len fields.
+      jlong offsets_addr = in_buf_addrs[buf_idx++];
+      jlong offsets_size = in_buf_sizes[sz_idx++];
+      auto offsets = std::shared_ptr<arrow::Buffer>(
+          new arrow::Buffer(reinterpret_cast<uint8_t*>(offsets_addr), offsets_size));
+      buffers.push_back(offsets);
+    if (arrow::is_binary_like(type->field(0)->type()->id())) {
+      // child offsets length is internal data length + 1
+      // offsets element is int32
+      // so here i just allocate extra 32 bit for extra 1 length
+      jlong offsets_addr = in_buf_addrs[buf_idx++];
+      jlong offsets_size = in_buf_sizes[sz_idx++];
+
+      auto child_offsets_buffer = std::shared_ptr<arrow::Buffer>(  new arrow::Buffer(reinterpret_cast<uint8_t*>(offsets_addr), offsets_size));
+    
+      buffers.push_back(std::move(child_offsets_buffer));
+    }
+  }
+
+  if (type->id() == arrow::Type::LIST) {
+    jlong offsets_addr = in_buf_addrs[buf_idx++];
```

Comment:
Does gandiva tests perform any memory leak checks? How do we make sure that added code does not result in memory leaks?